### ISSUE TITLE
magento/framework exception on grids

### ISSUE THREAD

**Issue by Morgy93** [baldwin-agency/magento2-module-url-data-integrity-checker#16]
Hello,

on `/baldwin_urldataintegritychecker/catalog_category_urlPath/` I'm getting the following error:

```
Call to undefined method Baldwin\UrlDataIntegrityChecker\Model\ResourceModel\Catalog\Category\UrlPathCollection\Interceptor::getSelect()
Exception in vendor/magento/framework/View/Element/UiComponent/DataProvider/FilterPool.php:45
```

All other grids are throwing the same exception.

Using version 1.2.0 of this module and Magento version 2.4.2 - maybe it is incompatible?

**Comment by hostep** [comment]
Hi @Morgy93

Indeed, I can reproduce this problem on a clean Magento 2.4.2 installation!

I've only tested this module on Magento 2.4.0 so far, not on 2.4.1 or 2.4.2, so something in those versions has changed which must have caused this problem.

I'll have a look later on for a way to solve this.

Thanks!

**Comment by hostep** [comment]
Magento 2.4.2 came with this change which starts causing this issue: https://github.com/magento/magento2/commit/2638e0ffae57f351c512e47a0295d4c2d376681a

This feels like a bug in Magento because a [Collection](https://github.com/magento/magento2/blob/2.4.2/lib/internal/Magento/Framework/Data/Collection.php) isn't being forced to implement a `getSelect()` method.

And since we use the Collection to read from a json file and not from the database this might be difficult to implement.

Possible solutions:
- start storing and reading data to/from the database instead of from a json file (which we were already planning of doing in scope of https://github.com/baldwin-agency/magento2-module-url-data-integrity-checker/issues/13#issuecomment-717997548)
- use an alternative Magento grid implementation like https://github.com/hyva-themes/magento2-hyva-admin (downsides: it currently only supports Magento 2.4 or higher and PHP 7.4 or higher)
- the combination of the two above
- doing something clever that is much less work then the above (not sure if that will be possible)

This might take a while until fixed since I don't have that much free time, but I'll see what I can do 🙂

**Comment by hostep** [comment]
Is fixed in version [1.2.1](https://github.com/baldwin-agency/magento2-module-url-data-integrity-checker/releases/tag/v1.2.1) of this module.

I picked option 4 and introduced a silly workaround for the issue. The other options would be better for future improvements, but would take a lot longer to implement.

Also, we aren't the only ones who run into this problem with Magento 2.4.2, see: https://github.com/magento/magento2/issues/32292

Thanks for the report @Morgy93!
